src/main/java/com/yelp/nrtsearch/server/luceneserver/search/query/multifunction/GeoPointDecayFilterFunction.java

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright 2024 Yelp Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.yelp.nrtsearch.server.luceneserver.search.query.multifunction;
+
+import com.google.type.LatLng;
+import com.yelp.nrtsearch.server.grpc.MultiFunctionScoreQuery;
+import com.yelp.nrtsearch.server.luceneserver.doc.LoadedDocValues;
+import com.yelp.nrtsearch.server.luceneserver.geo.GeoPoint;
+import com.yelp.nrtsearch.server.luceneserver.geo.GeoUtils;
+import java.io.IOException;
+import java.util.List;
+import java.util.Objects;
+import org.apache.lucene.index.*;
+import org.apache.lucene.search.Explanation;
+import org.apache.lucene.search.Query;
+
+public class GeoPointDecayFilterFunction extends DecayFilterFunction {
+
+  private final MultiFunctionScoreQuery.DecayFunction decayFunction;
+  private final String fieldName;
+  private final DecayFunction decayFunc;
+  private final double scale;
+  private final double offset;
+  private final double decay;
+  private final LatLng origin;
+
+  /**
+   * Constructor.
+   *
+   * @param filterQuery filter to use when applying this function, or null if none
+   * @param weight weight multiple to scale the function score
+   * @param decayFunction to score a document with a function that decays depending on the distance
+   *     between an origin point and a numeric doc field value
+   */
+  public GeoPointDecayFilterFunction(
+      Query filterQuery, float weight, MultiFunctionScoreQuery.DecayFunction decayFunction) {
+    super(filterQuery, weight, decayFunction);
+    this.decayFunction = decayFunction;
+    this.fieldName = decayFunction.getFieldName();
```

Comment:
If you also pass in the `IndexState` you get get the `FieldDef` for validation. The `FieldDef` can also be used to get the doc values accessor.